packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -306,11 +307,22 @@ export function createRootHooks<
     path: readonly string[],
     input: unknown,
     opts: UseTRPCSubscriptionOptions<unknown, TError>,
-  ) {
+  ): UseTRPCSubscriptionResult<unknown, TError> {
     const enabled = opts?.enabled ?? input !== skipToken;
     const queryKey = hashKey(getQueryKeyInternal(path, input, 'any'));
     const { client } = useContext();
 
+    const [subscriptionState, setSubscriptionState] = React.useState<
+      UseTRPCSubscriptionResult<unknown, TError>
+    >({
```

Comment:
Look at the types of React query - we should likely mimic those

Example

- error state is guaranteed to have error
- [...]